apps/ledger-live-mobile/src/logic/notifications.tsx

```diff
@@ -0,0 +1,386 @@
+    /** If set, we will not prompt the push notification modal again before this date unless the user triggers it manually from the settings */
+    dateOfNextAllowedRequest?: Date,
+    /** Whether or not the user clicked on the "Maybe later" cta */
+    alreadyDelayedToLater?: boolean,
+    /** If true, we will not prompt the push notification modal again unless the user triggers it manually from the settings */
+    doNotAskAgain?: boolean,
+};
+
+const pushNotificationsDataOfUserAsyncStorageKey = "pushNotificationsDataOfUser";
+
+const getCurrentRouteName = (
+    state: NavigationState | Required<NavigationState["routes"][0]>["state"],
+  ): Routes | undefined => {
+    if (state.index === undefined || state.index < 0) {
+      return undefined;
+    }
+    const nestedState = state.routes[state.index].state;
+    if (nestedState !== undefined) {
+      return getCurrentRouteName(nestedState);
+    }
+    return state.routes[state.index].name;
+};
+  
+async function getPushNotificationsDataOfUserFromStorage() {
+    const dataOfUser = await AsyncStorage.getItem(
+      pushNotificationsDataOfUserAsyncStorageKey,
+    );
+  
+    return JSON.parse(dataOfUser);
+}
+  
+async function setPushNotificationsDataOfUserInStorage(dataOfUser) {
+    await AsyncStorage.setItem(
+        pushNotificationsDataOfUserAsyncStorageKey,
+      JSON.stringify(dataOfUser),
+    );
+}
+  
+  
+const useNotifications = () => {
+    const pushNotificationsFeature = useFeature("pushNotifications");
+    const { pushToast } = useToasts();
+    const notificationsSettings = useSelector(notificationsSelector);
+
+    const [notificationsToken, setNotificationsToken] = useState();
+    const isPushNotificationsModalOpen = useSelector(notificationsModalOpenSelector);
+    const pushNotificationsModalType = useSelector(notificationsModalTypeSelector);
+    const pushNotificationsOldRoute = useSelector(notificationsCurrentRouteNameSelector);
+    const pushNotificationsEventTriggered = useSelector(notificationsEventTriggeredSelector);
+    const pushNotificationsDataOfUser = useSelector(notificationsDataOfUserSelector);
+    const accounts: Account[] = useSelector(accountsSelector);
+    const currAppLanguage = useSelector(languageSelector);
+
+    const accountsWithAmountCount = useMemo(() => accounts.filter(account => account.balance?.gt(0)).length, [accounts]);
+
+    const dispatch = useDispatch();
+    const navigation = useNavigation();
+
+    const getIsNotifEnabled = async () => {
+        const authStatus = await messaging().hasPermission();
+
+        return authStatus === messaging.AuthorizationStatus.AUTHORIZED ||
+        authStatus === messaging.AuthorizationStatus.PROVISIONAL;
+    };
+
+    const listenForNotifications = useCallback(async () => {
+        if (!notificationsToken) {
+            const token = await messaging().getToken();
+            setNotificationsToken(token);
+            messaging().onMessage(async remoteMessage => {
```

Comment:
i see a lot of `messaging().xxx`  does messaging() return an instance of FCM and wouldn't it be better to set a const in the component linking to it to not repeat this? :thinking: (maybe it will stop working idk)